### DISCUSSION TITLE
How to handle click on push-notification when application is in foreground?

### DISCUSSION THREAD

**Discussion by valerybodak** [firebase/flutterfire#7860] [Q&A]
I use the following version:
firebase_messaging: ^8.0.0-dev.15

I know that I can handle click on push-notification when app is in **background** in the following way:

```
FirebaseMessaging.onMessageOpenedApp.listen((RemoteMessage message) {
      //TODO
    });

```
and it works, but how to handle click on push-notification when application is in **foreground**? Can't find in the documentation as well. Could you please help me. Thanks in advance.

**Comment by Hemant-Gurjar** [comment]
this might help you 💯 --- https://youtu.be/f4Lo9u6Xgt8

although I didn't like the way he explained but kinda good
